Added Ajax request to get timelines from backend

HadarRose opened the pull request:
- tutorials used: https://stackoverflow.com/questions/42412006/serving-up-a-javascript-file-using-node-js

Commit:
Added Ajax request to get timelines from backend

kev-chan commented on src/index.html:
```diff
@@ -1,6 +1,13 @@
+            <p id="timeline-para"></p>
+        </div>
+
+        <script type="text/javascript" src="src/js/basicRequests.js"></script>
```
put this in the <head> tag

HadarRose replied:
I'm sorry, I'm not sure I understand which tag you are referring to, or what should be in it. Is the script supposed to be elsewhere?

kev-chan replied:
<head>

kev-chan replied:
head tag

kev-chan replied:
read up about what goes in the head tag and please adjust as necessary

kev-chan commented on src/js/basicRequests.js:
```diff
@@ -0,0 +1,11 @@
+    console.info('Requesting timeline from backend.');
+    let xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange = function(){ // more on readystates: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState
+        if(this.readyState == 4 && this.status == 200){ // waits for request to be done (readyState 4) and successful (status 200)
```
are there any existing variables you can use for the state or status?

HadarRose replied:
I can use XMLHttpRequest.DONE for the state (added, will be pushed with the other changes) but I can't find a variable for the status. Should I get it from a different source?

Commit:
Minor changes

- used variable instead of 4 for DONE
- changed body to head at index.html

Commit:
Fixed head and body at index

changed index so that only the script is in head, since that something pertaining the data that should not be displayed

kev-chan commented:
are all scenarios handled?

Commit:
added error handling

HadarRose commented:
@kev-chan added error handling, which I believe handles all scenarios that need handling.

kev-chan commented:
same comment as before

Commit:
Added bad status handling
